### PR TITLE
Fixes #4287 "Tapping multiple items in explore should not be allowed"

### LINKED ISSUE #4287: Tapping multiple items in explore should not be allowed
**Summary:** 
When tapping on multiple item at the same time the explore fragment is replaced by the media details.

**Steps to reproduce:** 
* Tap on multiple items in explore fragment simultaneously (use two fingers ;-) )
* Go back to the fragment from details ( back ).
* The fragment is replaced by a media detail.

**Device and Android version:** 
Nokis 6.1+ android 10 stock.
 
**Commons app version:** 
3.0.0 debug master.

**Would you like to work on the issue?**
YES.

### PR DESCRIPTION
**Pull request by 4D17Y4**
**Description (required)**

Fixes #4287

What changes did you make and why?

* Disabled opening multiple media details at a time.

**Tests performed (required)**

Tested  on Nokia 6.1+ with API level 29.

### CODE REVIEW

**Comment by 4D17Y4** [comment]
Up for review @nicolas-raoul , @neslihanturan

File: app/src/main/java/fr/free/nrw/commons/explore/ExploreListRootFragment.java [hunk 1]
```diff
@@ -31,6 +32,9 @@
   private MediaDetailPagerFragment mediaDetails;
   private CategoriesMediaFragment listFragment;
 
+  // true when an item can be clicked.
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/explore/ExploreListRootFragment.java:35]: Please use the javadoc convention format (the usual `/**`/etc multiline block)

File: app/src/main/java/fr/free/nrw/commons/explore/ExploreListRootFragment.java [hunk 2]
```diff
@@ -31,6 +32,9 @@
...
+  private boolean openMediaDetail=true;
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/explore/ExploreListRootFragment.java:36]: Spaces before and after `=`

File: app/src/main/java/fr/free/nrw/commons/explore/ExploreListRootFragment.java [hunk 3]
```diff
@@ -113,13 +117,27 @@ public void onAttach(final Context context) {
 
   @Override
   public void onMediaClicked(int position) {
-    Log.d("deneme8","on media clicked");
+
+    if(!openMediaDetail) {
+      // do not open Media detail if openMediaDetail is false
+      return;
+    }
+    openMediaDetail=false;
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/explore/ExploreListRootFragment.java:125]: spaces

File: app/src/main/java/fr/free/nrw/commons/explore/ExploreListRootFragment.java (hunk 2 above)
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/explore/ExploreListRootFragment.java:36]: openMediaDetail -> openMediaDetails

File: app/src/main/java/fr/free/nrw/commons/explore/ExploreListRootFragment.java [hunk 4]
```diff
@@ -113,13 +117,27 @@ public void onAttach(final Context context) {
...
+    Log.d("deneme8", "on media clicked");
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/explore/ExploreListRootFragment.java:126]: I guess you forgot to remove this line?

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]

**Comment by 4D17Y4** [comment]
There's another way to solve this issue by disabling any multitaps in the activity by setting them in the manifest file.
But as this may hinder some app functionalities I preferred this one.

Made requested changes. 
Thanks.

File: app/src/main/java/fr/free/nrw/commons/explore/ExploreListRootFragment.java [hunk 5]
```diff
@@ -113,13 +114,26 @@ public void onAttach(final Context context) {
...
     ((ExploreFragment) getParentFragment()).setScroll(false);
     setFragment(mediaDetails, listFragment);
+
+    // Handler which re-enables the click action on Media
+    new Handler().postDelayed(new Runnable() {
+      @Override
+      public void run() {
+        openMediaDetails = true;
+      }
+    }, 2000);
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/explore/ExploreListRootFragment.java:136]: I just tried, it fixes the problem.
A drawback is that when opening a picture's media details and pressing back and opening another picture's media details fast enough, the second opening does nothing. That might happen with people who do not know that swapping does the same.

Do you think there might be a better solution than this artificial delay? The scenario here (a list of items that open details) is a common pattern.

File: app/src/main/java/fr/free/nrw/commons/explore/ExploreListRootFragment.java [hunk 6]
```diff
@@ -113,13 +109,14 @@ public void onAttach(final Context context) {
 
   @Override
   public void onMediaClicked(int position) {
-    Log.d("deneme8","on media clicked");
+
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/explore/ExploreListRootFragment.java:112]: Did you purposefully add these new lines (at the beginning and end of the method)?
I don't think it is the usual convention, right?
By the way, you can easily check all of your changes by using Git's branch comparison tool.

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
I tried this version for normal usage today and tested various things, it worked as expected.

The source code looks OK too.

I requested the cancellation of a small formatting change, should be OK for merging after that.

File: app/src/main/java/fr/free/nrw/commons/explore/ExploreListRootFragment.java (hunk 6 above)
Comment (4D17Y4) [reply app/src/main/java/fr/free/nrw/commons/explore/ExploreListRootFragment.java:112]: I'll remove them,
I was trying code formatting tools and that's the reason for this unintended formatting.
I'll look into it, and take care to recheck before push.
Thanks

**Review by nicolas-raoul — APPROVED** [review]

**Pull request merged**